compiler/rustc_target/src/abi/call/x86_64.rs

```diff
@@ -153,9 +153,9 @@ fn reg_component(cls: &[Option<Class>], i: &mut usize, size: Size) -> Option<Reg
     }
 }
 
-fn cast_target(cls: &[Option<Class>], size: Size) -> Option<CastTarget> {
```

Comment:
Here we revert https://github.com/rust-lang/rust/commit/eddfce53c10d281cde6d283f36a12c43606b3f8c, which is no longer needed since we now reject the problematic type earlier.